grails-app/services/org/pih/warehouse/data/OutboundStockMovementDataService.groovy

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright (c) 2012 Partners In Health.  All rights reserved.
+ * The use and distribution terms for this software are covered by the
+ * Eclipse Public License 1.0 (http://opensource.org/licenses/eclipse-1.0.php)
+ * which can be found in the file epl-v10.html at the root of this distribution.
+ * By using this software in any fashion, you are agreeing to be bound by
+ * the terms of this license.
+ * You must not remove this notice, or any other, from this software.
+ **/
+package org.pih.warehouse.data
+
+import org.joda.time.LocalDate
+import org.pih.warehouse.core.Location
+import org.pih.warehouse.importer.ImportDataCommand
+import org.pih.warehouse.product.Product
+import org.pih.warehouse.requisition.Requisition
+import org.pih.warehouse.requisition.RequisitionItem
+import org.springframework.validation.BeanPropertyBindingResult
+import org.pih.warehouse.auth.AuthService
+
+class OutboundStockMovementDataService {
+
+    Boolean validateData(ImportDataCommand command) {
+        log.info "Validate data " + command.filename
+        command.data.eachWithIndex { params, index ->
+            if (!params?.origin) {
+                throw new IllegalArgumentException("Row ${index + 1}: Origin/Source is required")
+            }
+            if (!params?.destination) {
+                throw new IllegalArgumentException("Row ${index + 1}: Destination is required")
+            }
+            if (!params?.quantity) {
+                throw new IllegalArgumentException("Row ${index + 1}: Requested Quantity is required")
+            }
+            Requisition requisition = createOrUpdateRequisition(params)
+            if (!requisition.validate()) {
+                requisition.errors.each { BeanPropertyBindingResult error ->
+                    command.errors.reject("${index + 1}: ${requisition.name} ${error.getFieldError()}")
+                }
+            }
+        }
+    }
+
+    void importData(ImportDataCommand command) {
+        log.info "Import data " + command.filename
+        command.data.eachWithIndex {params, index ->
+            Requisition requisition = createOrUpdateRequisition(params)
+            if(requisition.validate()){
+                requisition.save(failOnError: true)
+            }
+        }
+    }
+
+    Requisition createOrUpdateRequisition(Map params ){
+        Product product = Product.findByProductCode(params.productCode)
+        if(!product) {
+            throw new IllegalArgumentException("Product not found for SKU Code")
+        }
+        def requisitionItem = RequisitionItem.createCriteria().get {
```

Comment:
This requires a bit more work. I think we should move this logic until after we find a requisition. And the uniqueness is probably a combination of requisition, product, and quantity.